internal/services/datafactory/data_factory_dataset_snowflake_resource.go

```diff
@@ -121,27 +121,49 @@ func resourceDataFactoryDatasetSnowflake() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeString,
 							Optional: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"Byte",
-								"Byte[]",
-								"Boolean",
-								"Date",
-								"DateTime",
-								"DateTimeOffset",
-								"Decimal",
-								"Double",
-								"Guid",
-								"Int16",
-								"Int32",
-								"Int64",
-								"Single",
-								"String",
-								"TimeSpan",
+								"NUMBER",
+								"DECIMAL",
+								"NUMERIC",
+								"INT",
+								"INTEGER",
+								"BIGINT",
+								"SMALLINT",
+								"FLOAT",
+								"FLOAT4",
+								"FLOAT8",
+								"DOUBLE",
+								"DOUBLE PRECISION",
+								"REAL",
+								"VARCHAR",
+								"CHAR",
+								"CHARACTER",
+								"STRING",
+								"TEXT",
+								"BINARY",
+								"VARBINARY",
+								"BOOLEAN",
+								"DATE",
+								"DATETIME",
+								"TIME",
+								"TIMESTAMP",
+								"TIMESTAMP_LTZ",
+								"TIMESTAMP_NTZ",
+								"TIMESTAMP_TZ",
+								"VARIANT",
+								"OBJECT",
+								"ARRAY",
+								"GEOGRAPHY",
 							}, false),
 						},
-						"description": {
```

Comment:
how come we are removing the description property?

Reply:
Seems that the description property is used only for the structure column (used in other resources) and not for the snowflake schema column. For every column in Snowflake the properties are name, type, precision and scale.

Reply:
ahh ok. as above removing this is a breaking change even if it currently does nothing, could we instead deprecate it and we'll remove it in 3.0 - thanks!